File: preppi_slim/SCR/run_elm.pl
#!/usr/bin/perl
use strict;
use warnings;

use MODS::Globals;
use MODS::Genome;
use MODS::Pipeline;
use MODS::Methods;
use Getopt::Std;

my $PGM = $0;                   #name of program
$PGM =~ s#.*/##;     
my $usage = <<USAGE;
USAGE:
        $PGM genome_name [-d] [-q]
        Parameters:
		
		-genome_name: Name of the genome. 
		-d Toggle on debug mode.
                -q Configure the pipeline without submitting it.
USAGE

if(@ARGV<1 ) {
  print STDERR $usage ;
  exit(-1);
  }

my $gname=shift;

my %opts;
getopts('pdqtf:m:a:',\%opts);
my $debug=$opts{d} ? 'yes' : 'no';
my $execute=$opts{q} ? 'no' : 'yes';

my $genome=new MODS::Genome(gname => $gname );

my $pipeline=new MODS::Pipeline(name=>"run_elm",gname=>$gname,debug=>$debug);
$pipeline->set_targets($opts{t}) if defined $opts{t};
unlink $pipeline->{stepsfn}, "$pipeline->{stepsfn}.focus";

# ELM protein peptide
$pipeline->add_step("FindMotifs_ELM");
$pipeline->add_step("FindPRDs_ELM");
$pipeline->add_step("Gopher");
$pipeline->add_step("MuscleG");
$pipeline->add_step("MotifConsv");

$pipeline->qsub($execute);

File: preppi_slim/SCR/run_elm.py
#!/usr/bin/env python3
import argparse
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[1]))

from MODS.Pipeline import Pipeline


def main():
    parser = argparse.ArgumentParser(description="Run PrePPI-SLiM annotation")
    parser.add_argument("genome")
    parser.add_argument("-d", "--debug", action="store_true")
    parser.add_argument("-q", "--no-submit", action="store_true")
    parser.add_argument("-t", "--targets")
    args = parser.parse_args()
    pipeline = Pipeline(name="run_elm", gname=args.genome, debug="yes" if args.debug else "no")
    Path(pipeline.stepsfn).unlink(missing_ok=True)
    Path(pipeline.stepsfn + ".focus").unlink(missing_ok=True)
    if args.targets:
        pipeline.set_targets(args.targets)
    for step in ("FindMotifs_ELM", "FindPRDs_ELM", "Gopher", "MuscleG", "MotifConsv"):
        pipeline.add_step(step)
    pipeline.qsub("no" if args.no_submit else "yes")


if __name__ == "__main__":
    main()
